plugins/api/src/touch.cpp

```diff
@@ -0,0 +1,541 @@
+            throw_if_less_than_zero<int>(contains_secs_since_epoch, opts, prop_seconds_since_epoch);
+        }
+    }
+
+    template <typename ExpectedType>
+    auto throw_if_incorrect_type(bool _check_value,
+                                 const json& _object,
+                                 const std::string_view _prop_name) -> void
+    {
+        if (!_check_value) {
+            return;
+        }
+
+        try {
+            _object.at(_prop_name.data()).get<ExpectedType>();
+        }
+        catch (const json::exception&) {
+            THROW(SYS_INVALID_INPUT_PARAM, fmt::format("Incorrect value for [{}].", _prop_name));
+        }
+    }
+
+    auto throw_if_empty_string(bool _check_value,
+                               const json& _object,
+                               const std::string_view _prop_name) -> void
+    {
+        if (!_check_value) {
+            return;
+        }
+
+        if (_object.at(_prop_name.data()).get<std::string>().empty()) {
+            THROW(INPUT_ARG_NOT_WELL_FORMED_ERR, fmt::format("[{}] cannot be empty.", _prop_name));
+        }
+    }
+
+    template <typename T>
+    auto throw_if_less_than_zero(bool _check_value,
+                                 const json& _object,
+                                 const std::string_view _prop_name) -> void
+    {
+        if (!_check_value) {
+            return;
+        }
+
+        if (_object.at(_prop_name.data()).get<T>() < 0) {
+            THROW(INPUT_ARG_NOT_WELL_FORMED_ERR, fmt::format("[{}] cannot be less than zero.", _prop_name));
+        }
+    }
+
+    auto get_time(rsComm_t& _comm, const json& _opts) -> fs::object_time_type
+    {
+        if (_opts.contains(prop_reference)) {
+            return fs::server::last_write_time(_comm, _opts.at(prop_reference.data()).get<std::string>());
+        }
+
+        using duration_type = fs::object_time_type::duration;
+
+        if (_opts.contains(prop_seconds_since_epoch)) {
+            return fs::object_time_type{duration_type{_opts.at(prop_seconds_since_epoch.data()).get<rep_type>()}};
+        }
+
+        return std::chrono::time_point_cast<duration_type>(fs::object_time_type::clock::now());
+    }
+
+    auto create_data_object_if_necessary(rsComm_t& _comm,
+                                         const json& _json_input,
+                                         const fs::path& _path) -> void
+    {
+        if (fs::server::exists(_comm, _path)) {
+            return;
+        }
+
+        // Get ready to create a new data object.
+        io::server::default_transport tp{_comm};
+
+        if (_json_input.contains(prop_options)) {
+            const auto& opts = _json_input.at(prop_options.data());
+
+            // Return immediately if the object does not exist and the user specified to not
+            // create any new data objects.
+            if (opts.contains(prop_no_create) && opts.at(prop_no_create.data()).get<bool>()) {
+                // TODO Use better error code.
```

Comment:
`USER_INCOMPATIBLE_PARAMS`?

Reply:
maybe this shouldn't be considered an error at all. `touch --no-create does_not_exist.txt` returns `0` which seems like the right thing to do.

Reply:
ah, very well.

`0` it is - no work was requested, and no work was done.